### PR TITLE
RWA Vue TS

### PR DESCRIPTION
**Pull request by GoodDayForSurf**

### CODE REVIEW

File: packages/vue/src/pages/crm-contact-form/crm-contact-form.vue [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                      [messages]="contactMessages"
+                  ></contact-cards>-->
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import {ref} from 'vue';
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-form/crm-contact-form.vue:76]: Add spaces

File: packages/vue/src/pages/crm-contact-form/crm-contact-form.vue [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+import DxButton from 'devextreme-vue/button';
+import {
+DxToolbar ,
+    DxItem as DxToolbarItem,
+} from "devextreme-vue/toolbar";
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-form/crm-contact-form.vue:82]: I think we need to replace these quotes with single quotes. And fix it in all application modules

File: packages/vue/src/pages/crm-contact-form/crm-contact-form.vue [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+const contactName = ref('');
+
+const refresh = () => {
+ loadData();
+};
+
+function loadData() {
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-form/crm-contact-form.vue:90]: Why do we have an empty function?

File: packages/vue/src/pages/crm-contact-list/api/crm-contact-service.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import http from '@/api/http';
+
+class CrmContactService {
+     getAll = () =>
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/api/crm-contact-service.ts:4]: I see that you use different indentations in different places. For axample here 4 spaces and here 2 spaces https://github.com/DevExpress/devextreme-ui-template-gallery/pull/58/files#diff-ba57406a9b57055921f60eaf5caa19f399b9eb0083685bd83e581c9794ef15c2R87

File: packages/vue/src/pages/crm-contact-list/contact-panel/components/form-item-editable.vue [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    <dx-text-box v-if="props.isEditing"
+                 :value="props.data[props.dataField]"
+                 :label="title"
+                 @value-changed="e => props.data[props.dataField] = e.value"
+    ></dx-text-box>
+</template>
+
+<script setup lang="ts">
+import {computed} from "vue";
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/contact-panel/components/form-item-editable.vue:15]: Add spaces

File: packages/vue/src/pages/crm-contact-list/contact-panel/components/form-item-editable.vue [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import FormItemBlue from '@/components/form-item-blue.vue'
+import {DxTextBox} from 'devextreme-vue/text-box';
+
+const props = withDefaults(
+    defineProps<{
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/contact-panel/components/form-item-editable.vue:21]: Strange indents

File: packages/vue/src/pages/crm-contact-list/contact-panel/contact-panel.vue [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+          </div>
+        </dx-scroll-view>
+      </template>
+
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import {onBeforeUnmount, onMounted, ref, watch} from 'vue';
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/contact-panel/contact-panel.vue:166]: Add spaces

File: packages/vue/src/pages/crm-contact-list/contact-panel/contact-panel.vue [hunk 8]
```diff
@@ -0,0 +1,401 @@
...
+  }).catch((e: string) => {
+    console.log(e);
+  });
+}
+
+function accordionPlusClick(/*e : Event*/){
+}
+
+watch(() => props.userId,
+    (newVal, _) => {
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/contact-panel/contact-panel.vue:253]: Fix indents

Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/contact-panel/contact-panel.vue:253]: Can you change the title to be more specific? For example newId or newUserId

File: packages/vue/src/pages/crm-contact-list/crm-contact-list.vue [hunk 9]
```diff
@@ -0,0 +1,255 @@
+<template>
+  <div class="view-wrapper">
+    <dx-data-grid ref="dataGrid"
+                  :data-source="gridData"
+                  height="100%"
+                  class="grid"
+                  @row-click="rowClick">
+
+      <!-- Options -->
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/crm-contact-list.vue:9]: Why did you add this comment?

File: packages/vue/src/pages/crm-contact-list/crm-contact-list.vue [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+      </template>
+
+      <template #statusCellTemplate="{ data }">
+        <user-status :status="data.data?.status"/>
+      </template>
+
+    </dx-data-grid>
+
+    <!--  Contact panel  -->
+    <contact-panel :userId="panelData?.id" :is-panel-open="isPanelOpen" @close="isPanelOpen = false"></contact-panel>
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/crm-contact-list.vue:115]: Fix the style of props. You use camelCase and kebab-case.

Comment (dxvladislavvolkov) [reply packages/vue/src/pages/crm-contact-list/crm-contact-list.vue:115]: Replace with kebab

File: packages/vue/src/pages/crm-contact-list/crm-contact-list.vue [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+  CrmContactService.getAll().then((response:any) => {
+    gridData.value = response.data;
+    dataGrid.value?.instance.endCustomLoading();
+  }).catch((e: string) => {
+    console.log(e);
+  });
+}
+
+const rowClick = (e: RowClickEvent) => {
+  if ( e.data.id ) {
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/crm-contact-list.vue:170]: Remove spaces inside brackets

File: packages/vue/src/pages/crm-contact-form/crm-contact-form.vue [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+            :options="{
+        text: 'Refresh',
+        icon: 'refresh',
+        onClick: refresh
+      }"
+        ></dx-toolbar-item>
+      </dx-toolbar>
+
+      <div class="panels">
+        <!--      <div class="left">
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/pages/crm-contact-form/crm-contact-form.vue:55]: it's WIP code

File: packages/vue/src/pages/crm-contact-list/crm-contact-list.vue (hunk 9 above)
Comment (GoodDayForSurf) [reply packages/vue/src/pages/crm-contact-list/crm-contact-list.vue:9]: I like to add such comments for many lines templates, i think it's more readable

**Review by MikeVitik — CHANGES_REQUESTED** [review]

File: packages/vue/cypress.json [hunk 13]
```diff
@@ -0,0 +1,3 @@
+{
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/cypress.json:1]: removed

File: packages/vue/package.json [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+    "test:unit": "vue-cli-service test:unit",
+    "test:e2e": "vue-cli-service test:e2e",
+    "lint": "vue-cli-service lint"
+  },
+  "dependencies": {
+    "axios": "^0.27.2",
+    "core-js": "^3.8.3",
+    "devextreme": "^22.1.4",
+    "devextreme-vue": "^22.1.3",
+    "rxjs": "^7.5.6",
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/package.json:17]: removed

File: packages/vue/src/api/http.ts [hunk 15]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios';
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/api/http.ts]: done

File: packages/vue/src/pages/crm-contact-form/crm-contact-form.vue [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+import ContactCards from './components/contact-cards/contact-cards.vue';
+
+const contactId = 12;
+
+const contactName = ref('');
+const contactData = ref<Contact | Record<string, unknown>>({});
+const isLoading = ref(false);
+const refreshNotifier = new Rx.Subject<void>();
+
+provide('refresh-notifier', refreshNotifier);
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/pages/crm-contact-form/crm-contact-form.vue:98]: ok

File: packages/vue/src/components/card-notes/card-notes.vue [hunk 17]
```diff
@@ -0,0 +1,146 @@
+<template>
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/components/card-notes/card-notes.vue:1]: it will be used on other pages

File: packages/vue/src/pages/crm-contact-form/components/contact-form.vue [hunk 18]
```diff
@@ -0,0 +1,282 @@
...
+  (isEditingValue) => {
+    isEditing.value = isEditingValue;
+  },
+);
+
+watch(
+  () => props.contactData,
+  (contactDataNew: Contact | null) => {
+    if (contactDataNew != null) {
+      contactDataSaved = structuredClone(contactDataNew);
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/pages/crm-contact-form/components/contact-form.vue:230]: rewrited

File: packages/vue/src/components/card-messages/card-messages.vue [hunk 19]
```diff
@@ -0,0 +1,201 @@
+<template>
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/components/card-messages/card-messages.vue:1]: it will be used in other pages

File: packages/vue/src/pages/crm-contact-form/components/contact-form.vue [hunk 20]
```diff
@@ -0,0 +1,282 @@
...
+      contactData.value = contactDataNew;
+    }
+  },
+);
+
+function toggleStylingMode() {
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/pages/crm-contact-form/components/contact-form.vue:236]: agree

File: packages/vue/src/router/index.ts [hunk 21]
```diff
@@ -0,0 +1,45 @@
+import { createRouter, createWebHashHistory } from 'vue-router';
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/router/index.ts:1]: agree

File: packages/vue/src/App.vue [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+  getScreenSizeInfo: { cssClasses: [] },
+});
+
+screen.getScreenSizeInfo = getScreenSizeInfo();
+
+function screenSizeChanged() {
+  screen.getScreenSizeInfo = getScreenSizeInfo();
+}
+
+onMounted(() => {
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/App.vue:38]: rewrited

File: packages/vue/src/components/icons/IconCommunity.vue [hunk 23]
```diff
@@ -0,0 +1,7 @@
+<template>
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/components/icons/IconCommunity.vue:1]: it's unused files. removed

File: packages/vue/src/App.vue [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+const vm = getCurrentInstance();
+
+const { title } = (vm?.proxy as unknown as {$appInfo: {title: string}}).$appInfo;
+const screen: {getScreenSizeInfo: ScreenSizeInfo} = reactive({
+  getScreenSizeInfo: { cssClasses: [] },
+});
+
+screen.getScreenSizeInfo = getScreenSizeInfo();
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/App.vue:32]: refactored

File: packages/vue/src/components/form-item-with-button.vue [hunk 25]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <dx-text-box
```
[the remark this answers is not in the document]
Comment (GoodDayForSurf) [reply packages/vue/src/components/form-item-with-button.vue:2]: cool

File: packages/vue/src/utils/media-query.ts [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+  return {
+    isXSmall: screenSizes['screen-x-small'],
+    isLarge: screenSizes['screen-large'],
+    cssClasses: Object.keys(screenSizes).filter((cl: string) => screenSizes[cl]),
+  };
+}
+
+export const screenInfo = ref(getScreenSizeInfo());
+
+export const subscribe = (handler:Handler) => handlers.add(handler);
```
Comment (MikeVitik) [inline packages/vue/src/utils/media-query.ts:44]: Remove unused subscribe, unsubscribe, handlers

File: packages/vue/src/pages/crm-contact-list/components/contact-panel.vue [hunk 27]
```diff
@@ -0,0 +1,414 @@
...
+  },
+  {
+    name: 'address',
+    editorOptions: {
+      stylingMode: editorOptions.value.stylingMode,
+      icon: 'home',
+    },
+  }];
+
+const screenSizeChanged = () => {
```
Comment (MikeVitik) [inline packages/vue/src/pages/crm-contact-list/components/contact-panel.vue:245]: Rewrite using watch screenInfo

**Review by MikeVitik — CHANGES_REQUESTED** [review]

File: packages/vue/src/components/contact-activities.vue [hunk 28]
```diff
@@ -0,0 +1,117 @@
...
+            </div>
+          </div>
+        </template>
+      </dx-list>
+    </load-component>
+  </div>
+</template>
+
+<script setup lang="ts">
+import DxList from 'devextreme-vue/list';
```
Comment (dxvladislavvolkov) [inline packages/vue/src/components/contact-activities.vue:29]: Use named import

**Review by MikeVitik — APPROVED** [review]

File: packages/vue/src/pages/crm-contact-form/components/contact-cards/contact-cards.vue [hunk 29]
```diff
@@ -0,0 +1,81 @@
...
+import { DxItem as DxTabItem } from 'devextreme-vue/tabs';
+import type { Task } from '@/types/task';
+import type { Activity } from '@/types/activities';
+import type { Message } from '@/types/messages';
+import ContactActivities from '@/components/contact-activities.vue';
+import CardNotes from '@/components/card-notes.vue';
+import CardMessages from '@/components/card-messages.vue';
+// eslint-disable-next-line import/no-unresolved
+import { getContactMessages } from 'dx-rwa-data';
+import CardTasks from './components/card-tasks.vue';
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-form/components/contact-cards/contact-cards.vue:46]: Before you use @ in path to module

File: packages/vue/src/pages/crm-contact-list/components/contact-panel.vue [hunk 30]
```diff
@@ -0,0 +1,392 @@
+<template>
+  <div id="contact-panel" class="panel" v-bind:class="{ pin: isPin, open: props.isPanelOpen }">
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/components/contact-panel.vue:2]: Remove v-bind

File: packages/vue/src/pages/crm-contact-list/components/contact-panel.vue [hunk 31]
```diff
@@ -0,0 +1,392 @@
...
+                        :label="'Assigned to'"
+                        :isEditing = "isEditing"
+                      />
+                    </dx-form-item>
+                  </dx-form-group-item>
+                </dx-form-group-item>
+
+                <dx-form-group-item>
+                  <dx-form-item v-for="(item, index) in underContactFields"
+                                v-bind:key="index"
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/components/contact-panel.vue:68]: Remove v-bind

File: packages/vue/src/pages/crm-contact-list/components/contact-panel.vue [hunk 32]
```diff
@@ -0,0 +1,392 @@
...
+            </div>
+            <div class="data-part">
+              <dx-accordion :multiple="true" :collapsible="true">
+                <template #title="{ data }">
+                  <span class="accordion-title">{{ data.title }}</span>
+                  <dx-button
+                    icon="add"
+                    type="default"
+                    stylingMode="text"
+                    @click="e => {e.event.stopPropagation(); accordionPlusClick(e);}"
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/components/contact-panel.vue:133]: Maybe create a method?

File: packages/vue/src/pages/crm-contact-list/components/contact-panel.vue [hunk 33]
```diff
@@ -0,0 +1,392 @@
...
+const isLoading = ref(false);
+const isPin = ref(false);
+const isPinEnabled = ref(true);
+const panelData = ref<Contact | null>(null);
+const props = withDefaults(defineProps<{
+  isPanelOpen: boolean,
+  userId: number | null
+}>(), { isPanelOpen: false, userId: null });
+
+const editorOptions = ref({ stylingMode: 'underlined' });
```
Comment (dxvladislavvolkov) [inline packages/vue/src/pages/crm-contact-list/components/contact-panel.vue:196]: for object use reactive
